spec/services/github_service.rb

```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+RSpec.describe GithubService do
+  subject { described_class.call(payload: payload, event: event) }
+
+  describe 'events' do
+    context 'pull request' do
+      let!(:payload) { (create :pull_request_payload).merge((create :repository_payload)) }
```

Comment:
`create :object` is used by FactoryBot to build a model and insert it in the database.

To build a model without inserting it in the database it uses `build :object`.

For payloads it would be the same because they are not ActiveRecords but to preserve the semantics of FactoryBot I would change `create` for `build`.

What do you think?

Reply:
I will refactor the payload factories to have a sub factory with an after_create which will merge the requested payload. Here an example:
```
    factory :pull_request_payload_with_repository do
      after(:create) do |pull_request_payload|
        pull_request_payload.merge!((create :repository_payload))
      end
    end
```

Reply:
Indeed, that is why I suggested using `build` instead. `:build` already has the meaning of building an object without persisting it into the database.

Somebody reading `build` in this method already knows that the `pull_request_payload` is not persisted without having to know the implementation details defined in a different place.